malcolm/modules/ADCore/parts/detectordriverpart.py

```diff
@@ -86,6 +148,25 @@ def configure(self,
         if self.is_hardware_triggered:
             # Start now if we are hardware triggered
             self.actions.arm_detector(context)
+        attribute_xml = self.build_attribute_xml()
+        self.attributes_filename = os.path.join(
+            fileDir, "%s-attributes.xml" % self.mri)
+        with open(self.attributes_filename, 'w') as xml:
+            xml.write(attribute_xml)
+
+        if hasattr(child, "attributesFile"):
+            attributes_filename = self.attributes_filename
+            if self.runs_on_windows:
+                error_msg = "No or multiple FilePathTranslatorPart found:" + \
```

Comment:
I would make translate_filepath into a classmethod that takes part_info and filename, then does the filter_single_value with error message in a single step. That would avoid the duplication in the other parts that use this Info